### DISCUSSION TITLE
Request:  Internationalization for Authentication Email Templates

### DISCUSSION THREAD

**Discussion by burggraf** [supabase/supabase#953] [Feature Requests] [answered]
### Update: refer to issue https://github.com/supabase/gotrue/issues/80 for the latest.

---

The Authentication Email Templates should be made available for i18n internationalization so you can send an email in any language based on the current language or locale setting.

So instead of having one set of templates, you'd have one set for each language your application supports.  I.e. "en", "es", "fr", "de".

To keep existing compatibility you could leave the existing templates alone as the "default templates" but override those with a language-specific version of a language or locale code is set.

**Comment by kiwicopple** [comment, marked as answer]
This is a great idea @burggraf ... Perhaps you can open this as a feature request in the GoTrue repository?
https://github.com/supabase/gotrue

**Reply by burggraf** [reply]
Done!

**Reply by activenode** [reply]
For the sake of completeness: https://github.com/supabase/gotrue/issues/80

**Comment by activenode** [comment]
I have a quick question related to this: Isn't almost any app using at least 2 or more languages? Hence shouldn't this be a feature that is one of the very most essentials?

How is the status here? 

Is it possible to programmatically change it since it is currently not possible in the UI? 

If this is not possible at all it would be a blocker for me since I need to support german language next to "en".

**Comment by inian** [comment]
We might add support for this, for now you can get the email urls directly from gotrue and handle the email templates in your own system. 

see this discussion for more https://github.com/supabase/supabase/discussions/2831#discussioncomment-1198236

**Reply by burggraf** [reply]
I've started a project related to this: [supabase-roll-your-own-auth]( https://github.com/burggraf/supabase-roll-your-own-auth).

**Reply by Inzeppelin** [reply]
> you can get the email urls directly from gotrue

@inian, ok, thnx, it might be a solution for now. But could you please tell how could I disable default email behavior in supabase.io? User might get two emails: first generated by default, and second from custom method.

**Reply by akkie** [reply]
Is there an answer to question from @Inzeppelin?

**Reply by leo-cheron** [reply]
This is an important topic for any international application, has this request been considered yet?

**Comment by kvetoslavnovak** [comment]
Here us the link to working solution: 
https://github.com/supabase/gotrue/issues/80#issuecomment-1507168133

**Comment by kulakowka** [comment]
Maybe this will help you: https://supabase.com/docs/guides/auth/auth-hooks/send-email-hook

**Reply by activenode** [reply]
yup, this pretty much gets the job done here

**Comment by code2fire00** [comment]
In case there are other people looking for a way to send auth emails depending on the user's language directly in the templayte:

When a user register, you can store his language locale in the metadata.
In the email templates you can access the user metadata and do conditional rendering depending on the language of the user.
Exemple:

```
{{ if eq .Data.preferre_lang "fr" }}
EMAIL IN FRENCH HERE
{{ else if eq .Data.preferred_lang "en" }}
EMAIL IN ENGLISH HERE
{{ end }}
```

**Reply by activenode** [reply]
Interesting solution. That works by default in the Supabase templates? 

The versatile solution though is using the Auth Hook https://supabase.com/docs/guides/auth/auth-hooks/send-email-hook

**Reply by z4122** [reply]
Can this way change the dafault email title? I have successfully changed the email content by this method.

**Reply by activenode** [reply]
> Can this way change the dafault email title? I have successfully changed the email content by this method.

Yes, the Auth Hook solution is done by sending your own email, so yes, you can change the title (referring to my answer)
